libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -430,33 +452,59 @@ func (segments *diskQueueSegments) sizeOnDisk() uint64 {
 type segmentReader struct {
 	src io.ReadSeekCloser
 	er  *EncryptionReader
+	cr  *CompressionReader
```

Comment:
Can you add short descriptions here and in `segmentWriter`, especially clarifying that `cr`/`cw` takes precedence over `er`/`ew` when present?